crates/env/src/engine/on_chain/impls.rs

```diff
@@ -209,6 +209,15 @@ impl EnvInstance {
 }
 
 impl EnvBackend for EnvInstance {
+    fn is_contract_storage(&mut self, key: &Key) -> bool {
+        let output = &mut self.scoped_buffer().take_rest();
+        match ext::get_storage(key.as_bytes(), output) {
```

Comment:
A seal operation `is_storage` would do us good here as well, same as in https://github.com/paritytech/ink/issues/507. Maybe it's time?